src/formatDate.js

```diff
@@ -7,8 +7,37 @@
  *
  * @returns {string}
  */
-function formatDate(date, fromFormat, toFormat) {
-  // write code here
+function formatDate(dateString, fromFormat, toFormat) {
+  const separator = fromFormat[3];
+  const newSeparator = toFormat[3];
+  const dateParts = dateString.split(separator);
+
+  const dateObj = {
+    YYYY: fromFormat.includes('YYYY')
+      ? dateParts[fromFormat.indexOf('YYYY')]
+      : null,
+    YY: fromFormat.includes('YY') ? dateParts[fromFormat.indexOf('YY')] : null,
+    MM: dateParts[fromFormat.indexOf('MM')],
+    DD: dateParts[fromFormat.indexOf('DD')],
+  };
+
+  if (fromFormat.includes('YY') && toFormat.includes('YYYY')) {
+    const year = +dateObj['YY'];
+
+    dateObj['YYYY'] = year < 30 ? `20${dateObj['YY']}` : `19${dateObj['YY']}`;
+  }
+
+  if (fromFormat.includes('YYYY') && toFormat.includes('YY')) {
+    dateObj['YY'] = dateObj['YYYY'].slice(-2);
+  }
+
+  const formattedDate = [
```

Comment:
You have hard-coded the indices (0, 1, 2) to access the toFormat array. While this works, it ties your implementation directly to the assumption of having three date parts. If the toFormat array were to change in the future, this code would break. A more flexible approach would be to loop over the toFormat array and build the formattedDate array dynamically, similar to the suggestion for building the dateObj.